package-lock.json

```diff
@@ -4242,6 +4245,11 @@
         "safer-buffer": "^2.0.2",
         "tweetnacl": "~0.14.0"
       },
+      "bin": {
+        "sshpk-conv": "bin/sshpk-conv",
+        "sshpk-sign": "bin/sshpk-sign",
+        "sshpk-verify": "bin/sshpk-verify"
+      },
```

Comment:
Did simply regenerating package-lock.json cause this property to be created? If so, sshpk may not have been installed correctly when it was originally added to the project.

Reply:
Yes, I didn't change anything manually in package-lock.json. Perhaps it wasn't being created with npm 6.x and it's an effect of switching to npm 7.x with `nvm use 12`?

Reply:
Hm, it's possible.

What version of npm did you use? I can't seem to actually replicate it.

Reply:
Last time package-lock.json was updated was in PR #35, but just running `npm install` on the current `main` branch after removing package-lock.json re-creates this section. It's totally possible that sshpk (which I'm not familiar with) wasn't installed correctly to begin with, but I don't know why previous runs of `npm install` didn't add this section before.

Maybe it's been brought in by some minor version update on some dependency? The dependency chain is:

```
- ├─┬ coveralls@3.1.0
│ └─┬ request@2.88.2
│   ├─┬ http-signature@1.2.0
│   │ └─┬ sshpk@1.16.1
```

coveralls has been at version 3.1.0 for a year now, request@2.88.2 is pretty old as well and http-signature@1.2.0 is 4 years old.

Reply:
I forgot that npm attempts to reconcile dependency versions with respect to those currently residing in the local node_modules folder. Once I removed that folder and attempted to install everything again, I had a new package-lock.json with that property.

Reply:
I think I found why.

If I just run
```
git checkout main
rm package-lock.json
npm install
```

That bin section is re-created, regardless of which npm version I am using. But if I run:

```
nvm use 10
rm package-lock.json
rm -rf node_modules
npm install
```

There isn't any bin section anymore. I think my node_modules might have been created with npm v7 (that comes with node v12), which leads to the creation of that section even if using an older npm to run `npm install` later.

I've pushed an update of a new, clean package-lock.json after removing `node_modules` and just running `npm install` once with npm v7.

Reply:
> There isn't any bin section anymore. I think my node_modules might have been created with npm v7 (that comes with node v12), which leads to the creation of that section even if using an older npm to run npm install later.

Yeah, npm seems to trust and reconcile what it already in node_modules with whatever is in package-lock.json, even if package-lock.json is outdated (like in this case). I did the exact thing as you.

Keep in mind that some Node.js versions allow for newer versions of npm. The latest version of Node.js 12 (`v12.22.1`) comes bundled with npm 6.x but that npm can be updated to the latest major version—7.x (which resolves to 7.13.0). Since `nvm use <version>` may or may not correspond to a specific version of npm, it's important to isolate its version as much as possible. In most cases, running `npx npm@<version> […]` will be enough to test various behaviors for any particular version of npm as long as the npx utility you're working with supports running npm in that way.

> I've pushed an update of a new, clean package-lock.json[…]

The new package-lock.json will definitely alleviate the reproducibility issues we're seeing here.